src/main/asciidoc/transactions.adoc

```diff
@@ -33,82 +41,257 @@ static class Config extends AbstractCouchbaseConfiguration {
+Can be used within a service implementation via direct instantiation with a transaction manager reference, or get prepared in an application context and passed to services as bean reference. Note: The transaction manager should always be configured as bean in the application context: in the first case given to the service directly, in the second case given to the prepared template.
+Supports setting the propagation behavior and the isolation level by name, for convenient configuration in context definitions.
+
+.Using the TransactionTemplate
+====
+[source,java]
+----
+// 5.0.0-M5 transaction template bean does not exist, the TransactionTemplate must be explicitly created.
+@Autowired TransactionManager transactionManager;
+TransactionTemplate transactionTemplate = new TransactionTemplate(transactionManager);
+
+// After M6 transaction template bean exists
+@Autowired TransactionTemplate transactionTemplate;
+
+    transactionTemplate.executeWithoutResult(status -> {
+        Person p = personOperations.findById(Person.class).one(id);
+        p.setFirstname(p.getFirstname()+" Junior");
+        personOperations.replaceById(Person.class).one(p);
+    });
+----
+====
+
+== Transactions with CouchbaseTransactionManager
+
+Note: Reactive Couchbase Transactions cannot be supported via the 
+@Transactional annotation. See the section "Transactions with CouchbaseTransactionalOperator"
+for supporting Reactive Couchbase Transactions using the Transactional Operator.
+ 
+@Transactional Describes a transaction attribute on an individual method or on a class.
+ 
+When this annotation is declared at the class level, it applies as a default
+to all methods of the declaring class and its subclasses. Note that it does not
+apply to ancestor classes up the class hierarchy; inherited methods need to be
+locally redeclared in order to participate in a subclass-level annotation. For
+details on method visibility constraints, consult the
+https://docs.spring.io/spring-framework/docs/current/reference/html/data-access.html#transaction[Transaction Management]
+section of the reference manual.
+ 
+This annotation is generally directly comparable to Spring's
+org.springframework.transaction.interceptor.RuleBasedTransactionAttribute
+class, and in fact AnnotationTransactionAttributeSource will directly
+convert this annotation's attributes to properties in RuleBasedTransactionAttribute,
+so that Spring's transaction support code does not have to know about annotations.
+ 
+===  Attribute Semantics
+ 
+If no custom rollback rules are configured in this annotation, the transaction
+will roll back on RuntimeException and Error but not on checked
+exceptions.
+ 
+Rollback rules determine if a transaction should be rolled back when a given
```

Comment:
Yeah I don't think we implemented any of this...

Reply:
We _could_, but I'd probably just document that rollback rules aren't supported, and see if we get any requests for it.

Reply:
For my reference, this is now resolved.